src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/TypeAccessibilityCodeActionProvider.cs

```diff
@@ -55,169 +59,158 @@ internal class TypeAccessibilityCodeActionProvider : CSharpCodeActionProvider
-                if (codeAction.Title.StartsWith(expectedCodeActionPrefix, StringComparison.OrdinalIgnoreCase))
-                {
-                    fqn = codeAction.Title.Substring(expectedCodeActionPrefix.Length);
+                    var addUsingCodeAction = AddUsingsCodeActionProviderFactory.CreateAddUsingCodeAction(fqn, context.Request.TextDocument.Uri);
+                    if (addUsingCodeAction != null)
+                    {
+                        typeAccessibilityCodeActions.Add(addUsingCodeAction);
+                    }
                 }
             }
 
-            if (string.IsNullOrEmpty(fqn))
-            {
-                typeAccessibilityCodeActions = default;
-                return false;
-            }
-
-            typeAccessibilityCodeActions = new List<RazorCodeAction>();
-
-            var fqnCodeAction = CreateFQNCodeAction(context, diagnostic, codeAction, fqn);
-            typeAccessibilityCodeActions.Add(fqnCodeAction);
-
-            var addUsingCodeAction = AddUsingsCodeActionProviderFactory.CreateAddUsingCodeAction(fqn, context.Request.TextDocument.Uri);
-            if (addUsingCodeAction != null)
-            {
-                typeAccessibilityCodeActions.Add(addUsingCodeAction);
-            }
-
-            return true;
+            return typeAccessibilityCodeActions;
         }
 
-        private static bool TryProcessCodeActionVS(
+        private static IEnumerable<RazorCodeAction> ProcessCodeActionsVS(
             RazorCodeActionContext context,
-            RazorCodeAction codeAction,
-            Diagnostic diagnostic,
-            out ICollection<RazorCodeAction> typeAccessibilityCodeActions)
+            IEnumerable<RazorCodeAction> codeActions)
         {
-            typeAccessibilityCodeActions = new List<RazorCodeAction>(1);
+            var typeAccessibilityCodeActions = new List<RazorCodeAction>(1);
 
-            if (codeAction.Name.Equals(RazorPredefinedCodeFixProviderNames.FullyQualify, StringComparison.Ordinal))
+            foreach (var codeAction in codeActions)
             {
-                if (codeAction.Children?.Length == 0)
+                if (codeAction.Name.Equals(RazorPredefinedCodeFixProviderNames.FullyQualify, StringComparison.Ordinal))
                 {
-                    // When there's only one FQN suggestion, code action title is of the form:
-                    // `System.Net.Dns`
-                    var fqn = codeAction.Title;
-                    typeAccessibilityCodeActions.Add(CreateFQNCodeAction(context, diagnostic, codeAction, fqn));
+                    // The formatting pass of our Default code action resolver rejects
+                    // implicit/explicit expressions. So if we're in an implicit/explicit expression, 
+                    // we run the remapping resolver responsible for simply remapping 
+                    // (without formatting) the resolved code action.
+                    var action = IsImplicitOrExplicitExpression() ?
```

Comment:
I'm not a fan of local functions like this that make use of local variables. It's hard to understand what it's operating on. I'd suggest making the local function static and re-passing in the parameters that are necessary